patcher/patcher.go

```diff
@@ -34,6 +35,106 @@ func min(a uint32, b uint32) uint32 {
 	return a
 }
 
+func max(a uint32, b uint32) uint32 {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func SearchMemory(proc windows.Handle, LPBaseOfDll uintptr, SizeOfImage uint32, value []byte, altvalue []byte) (uintptr, error) {
+	// Information about the contents of the memory to read
+	var memoryBasicInfo struct {
+		BaseAddress       uintptr
+		AllocationBase    uintptr
+		AllocationProtect uint32
+		PartitionId       uint16
+		RegionSize        uintptr
+		State             uint32
+		Protect           uint32
+		Type              uint32
+	}
+
+	var memoryBasicInfoSize = uint32(unsafe.Sizeof(memoryBasicInfo))
+
+	var p uintptr = LPBaseOfDll
+	var offset uintptr
+	var MAX_CHUNK_SIZE uintptr = 4096 // Just an abitrary size
```

Comment:
Is there a reason we don't just load the entire image at once and search that way? Doing this with chunks and having to handle cross-chunk instances just seems like it's making the code overly complicated for no reason. Patching is only done once so I don't see a big disadvantage if it takes slightly longer with loading the entire image at once.

Reply:
That's true for totsugeki, though I still like this more in general since this is more generic for any amount of memory (not just the image).

Reply:
I'm a huge sucker for generic reusable code, even if it's overkill for the task at hand.

But I agree that in a professional setting, this is sort of over engineered.